staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy.go

```diff
@@ -259,3 +186,103 @@ func MatchCustomResourceDefinition(label labels.Selector, field fields.Selector)
 func CustomResourceDefinitionToSelectableFields(obj *apiextensions.CustomResourceDefinition) fields.Set {
 	return generic.ObjectMetaFieldsSet(&obj.ObjectMeta, true)
 }
+
+func dropDisabledFields(crdSpec, oldCrdSpec *apiextensions.CustomResourceDefinitionSpec) {
+	// if the feature gate is disabled, drop the feature.
+	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceValidation) {
+		unversionedValidationInUse := oldCrdSpec != nil && oldCrdSpec.Validation != nil
+		versionedValidationInUse := versionSchemaInUse(oldCrdSpec)
```

Comment:
since `versionSchemaInUse` only has a single caller, let's change it to `validationInUse` and make it do both the top-level and per-version check, so this block can become:

```
if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceValidation) && !validationInUse(oldCrdSpec) {
```